exchanges/orderbook/depth_test.go

```diff
@@ -142,11 +142,21 @@ func TestFlush(t *testing.T) {
 func TestUpdateBidAskByPrice(t *testing.T) {
 	d := newDepth(id)
 	d.LoadSnapshot(Items{{Price: 1337, Amount: 1, ID: 1}}, Items{{Price: 1337, Amount: 10, ID: 2}}, 0, time.Time{}, false)
-	d.UpdateBidAskByPrice(Items{{Price: 1337, Amount: 2, ID: 1}}, Items{{Price: 1337, Amount: 2, ID: 2}}, 0, 0, time.Time{})
+
+	// empty
+	d.UpdateBidAskByPrice(nil, nil, 0, 1, time.Time{})
+
+	d.UpdateBidAskByPrice(Items{{Price: 1337, Amount: 2, ID: 1}}, Items{{Price: 1337, Amount: 2, ID: 2}}, 0, 1, time.Time{})
+	if d.Retrieve().Asks[0].Amount != 2 || d.Retrieve().Bids[0].Amount != 2 {
+		t.Fatal("orderbook amounts not updated correctly")
+	}
+	// This should not do anything, as it simulates a websocket update
```

Comment:
This scenario doesn't work with the latest changes. Since the price and amount are the same anyway the test doesn't fail. But if you change the line to 
`d.UpdateBidAskByPrice(Items{{Price: 131337, Amount: 3, ID: 1}}, Items{{Price: 131237, Amount: 34, ID: 2}}, 0, 1, time.Time{})`
You see what I mean